Tests/WitsmlExplorer.Api.Tests/Workers/CopyWbGeometrySectionsWorkerTests.cs

```diff
@@ -0,0 +1,148 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Logging;
+
+using Moq;
+
+using Witsml;
+using Witsml.Data;
+using Witsml.ServiceReference;
+
+using WitsmlExplorer.Api.Jobs;
+using WitsmlExplorer.Api.Jobs.Common;
+using WitsmlExplorer.Api.Models;
+using WitsmlExplorer.Api.Services;
+using WitsmlExplorer.Api.Workers;
+using WitsmlExplorer.Api.Workers.Copy;
+
+using Xunit;
+
+namespace WitsmlExplorer.Api.Tests.Workers
+{
+    public class CopyWbGeometrySectionsWorkerTests
+    {
+        private readonly CopyWbGeometrySectionsWorker _copyWbGeometrySectionWorker;
+        private readonly Mock<IWitsmlClient> _witsmlClient;
+        private const string WellUid = "wellUid";
+        private const string WellboreUid = "sourceWellboreUid";
+        private const string TargetWbGeometryUid = "targetWbGeometryUid";
+        private const string SourceWbGeometryUid = "sourceWbGeometryUid";
+        private const string Uid1 = "Uid1";
+        private const string Uid2 = "Uid2";
+        private const string Uid3 = "Uid3";
+        private const string Uid4 = "Uid4";
+
+        public CopyWbGeometrySectionsWorkerTests()
+        {
+            Mock<IWitsmlClientProvider> witsmlClientProvider = new();
+            _witsmlClient = new Mock<IWitsmlClient>();
+            witsmlClientProvider.Setup(provider => provider.GetClient()).Returns(_witsmlClient.Object);
+            witsmlClientProvider.Setup(provider => provider.GetSourceClient()).Returns(_witsmlClient.Object);
+            Mock<ILogger<CopyWbGeometrySectionsJob>> logger = new();
+            _copyWbGeometrySectionWorker = new CopyWbGeometrySectionsWorker(logger.Object, witsmlClientProvider.Object);
+        }
+
+        [Fact]
+        public async Task CopyWbGeometrySection_TwoComponentsToCopy_TwoInQuery()
+        {
+            CopyWbGeometrySectionsJob copyWbGeometrySectionJob = CreateJobTemplate();
+            _witsmlClient.Setup(client =>
+                    client.GetFromStoreAsync(It.Is<WitsmlWbGeometrys>(witsmlWbGeometrys => witsmlWbGeometrys.WbGeometrys.First().Uid == SourceWbGeometryUid), new OptionsIn(ReturnElements.All, null)))
+                .ReturnsAsync(GetSourceWbGeometrys());
+            _witsmlClient.Setup(client =>
+                    client.GetFromStoreAsync(It.Is<WitsmlWbGeometrys>(witsmlWbGeometrys => witsmlWbGeometrys.WbGeometrys.First().Uid == TargetWbGeometryUid), new OptionsIn(ReturnElements.All, null)))
+                .ReturnsAsync(GetTargetWbGeometrys());
+            List<WitsmlWbGeometrys> copyWbGeometrySectionQuery = SetupUpdateInStoreAsync();
+
+            (WorkerResult, RefreshAction) result = await _copyWbGeometrySectionWorker.Execute(copyWbGeometrySectionJob);
+            WitsmlWbGeometry updatedWbGeometry = copyWbGeometrySectionQuery.First().WbGeometrys.First();
+            Assert.True(result.Item1.IsSuccess);
```

Comment:
The success result of `Execute` is the only thing we test for, but I think it could beneficial to also have test methods for the two error situations in the `Execute` method, what do you think?